socorro/external/es/crashstorage.py

```diff
@@ -115,29 +116,24 @@ def is_valid_key(key):
     return bool(VALID_KEY.match(key))
 
 
-def remove_bad_keys(data):
-    """Removes keys from the top-level of the dict that are bad
+def extract_indexable(fields, namespace, doc):
+    """Builds a doc of all the keys that should be indexed
 
-    Good keys satisfy the following properties:
+    This copies the data over, so the new doc is not tied to the original document at
+    all and can be mutated safely.
 
-    * have one or more characters
-    * are composed of a-zA-Z0-9_-
-
-    Anything else is a bad key and needs to be removed.
-
-    Note: This modifies the data dict in-place and only looks at the top level.
+    """
+    new_doc = {}
 
-    :arg dict data: the data to remove bad keys from
+    for field in FIELDS.values():
+        if field["namespace"] != namespace:
+            continue
 
-    """
-    if not data:
-        return
+        key = field["in_database_name"]
+        if key in doc:
+            new_doc[key] = copy.deepcopy(doc[key])
```

Comment:
We need the deepcopy for `json_dump`. That should be the only field that's deep. The rest of them are strings and ints and this should be a fast operation.